73_ArithmeticBench/main.cpp

```diff
@@ -0,0 +1,949 @@
+		}
+
+		// create dummy image
+		dummyImg = m_device->createImage({
+				{
+					.type = IGPUImage::ET_2D,
+					.samples = asset::ICPUImage::ESCF_1_BIT,
+					.format = asset::EF_R16G16B16A16_SFLOAT,
+					.extent = {WIN_W, WIN_H, 1},
+					.mipLevels = 1,
+					.arrayLayers = 1,
+					.flags = IImage::ECF_NONE,
+					.usage = core::bitflag(asset::IImage::EUF_STORAGE_BIT) | asset::IImage::EUF_TRANSFER_SRC_BIT
+				}
+			});
+		if (!dummyImg || !m_device->allocate(dummyImg->getMemoryReqs(), dummyImg.get()).isValid())
+			return logFail("Could not create HDR Image");
+
+		// create Descriptor Sets and Pipeline Layouts
+		smart_refctd_ptr<IGPUPipelineLayout> benchPplnLayout;
+		{
+			// create Descriptor Set Layout
+			smart_refctd_ptr<IGPUDescriptorSetLayout> dsLayout;
+			{
+				IGPUDescriptorSetLayout::SBinding binding[2];
+				for (uint32_t i = 0u; i < 2; i++)
+					binding[i] = {{},i,IDescriptor::E_TYPE::ET_STORAGE_BUFFER,IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,IShader::E_SHADER_STAGE::ESS_COMPUTE,1u,nullptr };
+				binding[1].count = OutputBufferCount;
+				dsLayout = m_device->createDescriptorSetLayout(binding);
+			}
+
+			// set and transient pool
+			auto descPool = m_device->createDescriptorPoolForDSLayouts(IDescriptorPool::ECF_NONE,{&dsLayout.get(),1});
+			testDs = descPool->createDescriptorSet(smart_refctd_ptr(dsLayout));
+			{
+				IGPUDescriptorSet::SDescriptorInfo infos[1+OutputBufferCount];
+				infos[0].desc = gpuinputDataBuffer;
+				infos[0].info.buffer = { 0u,gpuinputDataBuffer->getSize() };
+				for (uint32_t i = 1u; i <= OutputBufferCount; i++)
+				{
+					auto buff = outputBuffers[i - 1];
+					infos[i].info.buffer = { 0u,buff->getSize() };
+					infos[i].desc = std::move(buff); // save an atomic in the refcount
+				}
+
+				IGPUDescriptorSet::SWriteDescriptorSet writes[2];
+				for (uint32_t i=0u; i<2; i++)
+					writes[i] = {testDs.get(),i,0u,1u,infos+i};
+				writes[1].count = OutputBufferCount;
+
+				m_device->updateDescriptorSets(2, writes, 0u, nullptr);
+			}
+			testPplnLayout = m_device->createPipelineLayout({}, std::move(dsLayout));
+
+
+			smart_refctd_ptr<IGPUDescriptorSetLayout> benchLayout;
+			{
+				IGPUDescriptorSetLayout::SBinding binding[3];
+				for (uint32_t i = 0u; i < 2; i++)
+					binding[i] = { {},i,IDescriptor::E_TYPE::ET_STORAGE_BUFFER,IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,IShader::E_SHADER_STAGE::ESS_COMPUTE,1u,nullptr };
+				binding[1].count = OutputBufferCount;
+				binding[2] = { {},2,IDescriptor::E_TYPE::ET_STORAGE_IMAGE,IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_UPDATE_AFTER_BIND_BIT,IShader::E_SHADER_STAGE::ESS_COMPUTE,1u,nullptr };
+				benchLayout = m_device->createDescriptorSetLayout(binding);
+			}
+
+			benchPool = m_device->createDescriptorPoolForDSLayouts(IDescriptorPool::ECF_UPDATE_AFTER_BIND_BIT, { &benchLayout.get(),1 });
+			benchDs = benchPool->createDescriptorSet(smart_refctd_ptr(benchLayout));
```

Comment:
delete testing code, or use one descriptor layout and pipeline layout to rule them all